content/general/concepts/abstract-factory/abstract-factory.md

```diff
@@ -0,0 +1,394 @@
+| Bronze        | Max overdraft limit of 500  | Interest rate of 1.5% |
+| Builder       | No overdraft                | Interest rate of 1.5% |
+
+To simulate requesting and receiving a customer's credit score, a gateway has been mocked below. When given a customer's name, the`CreditAgencyGateway` class should return an appropriate `Customer` object. We can later use this class to see the different paths through our abstract factory.
+
+```java
+public class CreditAgencyGateway {
+  // Depending on which name is searched for, a different customer object is returned
+  public Customer getCustomer(String name) {
+    return switch (name) {
+      case "Harry" -> new Customer("Harry",
+                                   "Potter",
+                                   LocalDate.of(1980, 7, 31),
+                                   "4 Privet Drive, Little Whinging",
+                                   795);
+      case "Ron" -> new Customer("Ron",
+                                 "Weasley",
+                                 LocalDate.of(1980, 3, 1),
+                                 "The Burrow, Devon",
+                                 379);
+      case "Hermione" -> new Customer("Hermione",
+                                      "Granger",
+                                      LocalDate.of(1979, 9, 19),
+                                      "Hampstead Garden Suburb, London",
+                                      843);
+      default -> throw new IllegalArgumentException("Could not return the credit history for " + name);
+    };
+  }
+}
+```
+
+The `CreditAgencyGateway` class uses a `switch` statement to query a provided `name`. If the name is recognized, a new customer is returned. Otherwise, an exception is thrown.
+
+One of the main advantages of factory patterns is they allow for a large amount of model classes and enforce a common interface between them. In the code below, an abstract model `CurrentAccount` class works as an extension for its concrete sub-classes to be based on:
+
+```java
+public abstract class CurrentAccount {
+  private final String accountUID = UUID.randomUUID().toString();
+  private final Customer accountHolder;
+  private int balance;
+  private int agreedOverdraftLimit;
+
+  public CurrentAccount(Customer accountHolder, int initBalance, int agreedOverdraftLimit) {
+    this.accountHolder = accountHolder;
+    this.balance = initBalance;
+    this.agreedOverdraftLimit = agreedOverdraftLimit;
+  }
+
+  public void addFunds(final int amount) {
+    balance += amount;
+  }
+
+  public void withdrawFunds(final int amount) {
+    if (amount < balance + agreedOverdraftLimit) {
+      balance -= amount;
+    } else {
+      System.out.println("Insufficient funds. Unable to withdraw " + amount);
+    }
+  }
+
+  // Enforces concrete classes to override this method
+  public abstract void increaseOverdraft(int requestedLimit);
+
+  // Getters
+}
+```
+
+**Note:** Getters have been excluded from the above code snippet.
```

Comment:
```suggestion
For brevity, the getters have been excluded from the code snippet above.
```

Reply:
Hay @Dusch4593 - I seem to be getting some contradictory advice at the min. I think @yangc95 wanted these notes about getters to be removed as it will make it the same to other entries. I'm super happy to go either way, but would want consistency within the entry. Please see the comment on line 214.

Please advise?

Reply:
Ive removed to bring inline. Happy to re open should you feel its needed :)